deepchecks/vision/datasets/detection/coco_tensorflow.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module for loading a sample of the COCO dataset and the yolov5s model."""
+try:
+    import tensorflow as tf  # noqa: F401
+    import tensorflow_hub as hub  # noqa: F401
+except ImportError as error:
+    raise ImportError('tensorflow or tensorflow_hub is not installed. Please install them '
+                      'in order to use tensorflow coco dataset.') from error
+
+import os
+import typing as t
+from pathlib import Path
+
+import albumentations as A
+import numpy as np
+from typing_extensions import Literal
+
+from deepchecks import vision
+from deepchecks.vision.datasets.detection.coco_utils import COCO_DIR, LABEL_MAP, download_coco128, get_image_and_label
+from deepchecks.vision.vision_data import VisionData
+
+__all__ = ['load_dataset']
+
+_MODEL_URL = 'https://tfhub.dev/tensorflow/ssd_mobilenet_v2/fpnlite_320x320/1'
+TRAIN_FRACTION = 0.5
+PROBA_THRESHOLD = 0.5
+
+
+def load_dataset(
+        train: bool = True,
+        shuffle: bool = False,
+        object_type: Literal['VisionData', 'Dataset'] = 'Dataset',
+        n_samples: t.Optional[int] = None,
+) -> t.Union[tf.data.Dataset, vision.VisionData]:
+    """Get the COCO128 dataset and return a dataloader.
+
+    Parameters
+    ----------
+    train : bool, default: True
+        if `True` train dataset, otherwise test dataset
+    shuffle : bool, default: False
+        Whether to shuffle the dataset.
+    object_type : Literal['Dataset', 'DataLoader'], default: 'DataLoader'
+        type of the return value. If 'Dataset', :obj:`deepchecks.vision.VisionDataset`
+        will be returned, otherwise :obj:`tf.data.Dataset`.
+    n_samples : int, optional
+        Number of samples to load. Return the first n_samples if shuffle
+        is False otherwise selects n_samples at random. If None, returns all samples.
+
+    Returns
+    -------
+    Union[Dataset, VisionData]
+        A Dataset or VisionData instance representing COCO128 dataset
+    """
+    transforms = A.Compose([A.NoOp()], bbox_params=A.BboxParams(format='coco'))
+    coco_dataset = create_tf_dataset(train, n_samples, transforms)
+    if shuffle:
+        coco_dataset = coco_dataset.shuffle(128)
+
+    if object_type == 'Dataset':
+        return coco_dataset
+    elif object_type == 'VisionData':
+        model = hub.load(_MODEL_URL)
+        coco_dataset = coco_dataset.map(deepchecks_map(model))
+        return VisionData(batch_loader=coco_dataset, label_map=LABEL_MAP, task_type='object_detection',
+                          shuffle_batch_loader=False)
```

Comment:
What happens if user sets that to True for TF?

Reply:
Didnt understand